apps/test/unit/code-studio/initApp/projectTest.js

```diff
@@ -41,3 +43,34 @@ describe('project.getProjectUrl', function () {
     expect(project.getProjectUrl('/view')).to.equal('http://url/view?query');
   });
 });
+
+describe('project.saveThumbnail', () => {
+  const STUB_CHANNEL_ID = 'STUB-CHANNEL-ID';
+  let updateTimestamp;
+
+  beforeEach(() => {
+    updateTimestamp = sinon.stub(header, 'updateTimestamp').callsFake(() => {});
+    const projectData = {
+      id: STUB_CHANNEL_ID,
+      isOwner: true
+    };
+    project.updateCurrentData_(null, projectData);
+  });
+
+  afterEach(() => {
+    updateTimestamp.restore();
+  });
+
+  it.only('calls filesApi.putFile with correct parameters', () => {
+    const blob = 'stub-binary-data';
+    const putFile = sinon.stub(filesApi, 'putFile');
```

Comment:
It may be preferable to stub `putFile` in `beforeEach` - otherwise we might fail to restore the stub if an error occurs inside `project.saveThumbnail()`.  Also, there's no need to save a reference to it because `filesApi.putFile` gets replaced with the stub in this call - you can call `filesApi.putFile.restore()` when you're done.